Tribler/community/gigachannel/community.py

```diff
@@ -82,20 +98,103 @@ def on_blob(self, peer, blob):
+        self._logger.info("Started remote search for query:%s", query_filter)
+
+        for peer in search_candidates:
+            self.endpoint.send(peer.address, self.ezr_pack(self.SEARCH_REQUEST, search_request_payload))
+
+    @lazy_wrapper(SearchRequestPayload)
+    def on_search_request(self, peer, request):
+        # Caution: SQL injection
+        # Since this string 'query_filter' is passed as it is to fetch the results, there could be a chance for
+        # SQL injection. But since we use pony which is supposed to be doing proper variable bindings, it should
+        # be relatively safe
+        query_filter = request.query_filter.decode('utf8')
+        # Check if the query_filter is a simple query
+        if not is_simple_match_query(query_filter):
+            self.logger.error("Dropping a complex remote search query:%s", query_filter)
+            return
+
+        metadata_type = {'': [REGULAR_TORRENT, CHANNEL_TORRENT],
+                         "channel": CHANNEL_TORRENT,
+                         "torrent": REGULAR_TORRENT
+                        }.get(request.metadata_type, REGULAR_TORRENT)
+
+        request_dict = {
+            "first": 1,
+            "last": max_entries,
+            "sort_by": request.sort_by,
+            "sort_asc": request.sort_asc,
+            "query_filter": query_filter,
+            "hide_xxx": request.hide_xxx,
+            "metadata_type": metadata_type,
+            "exclude_legacy": True
+        }
+
+        result_blob = None
+        with db_session:
+            db_results, total = self.metadata_store.TorrentMetadata.get_entries(**request_dict)
+            if total > 0:
+                result_blob = entries_to_chunk(db_results[:max_entries], maximum_payload_size)[0]
+        if result_blob:
+            self.endpoint.send(peer.address, self.ezr_pack(self.SEARCH_RESPONSE,
+                                                           SearchResponsePayload(request.id, result_blob)))
+
+    @lazy_wrapper(SearchResponsePayload)
+    def on_search_response(self, peer, response):
+        search_request_cache = self.request_cache.get(u"remote-search-request", response.id)
+        if not search_request_cache or not search_request_cache.process_peer_response(peer):
+            return
+
+        with db_session:
+            try:
+                metadata_result = self.metadata_store.process_compressed_mdblob(response.raw_blob)
+            except (TransactionIntegrityError, CacheIndexError) as err:
+                self._logger.error("DB transaction error when tried to process search payload: %s", str(err))
+                return
+
+            search_results = [(dict(type={REGULAR_TORRENT: 'torrent', CHANNEL_TORRENT: 'channel'}[r.metadata_type],
+                                    **(r.to_simple_dict()))) for (r, _) in metadata_result
+                              if r and (r.metadata_type == CHANNEL_TORRENT or r.metadata_type == REGULAR_TORRENT)]
```

Comment:
these lines should be changed to: 
```python
                                    **(r.to_simple_dict()))) for (r, action) in metadata_result
                              if r and (a in [UPDATED_OUR_VERSION, UNKNOWN_TORRENT, UNKNOWN_CHANNEL])]
```
otherwise, the peer can send you DELETE entry, and then `process_payload` will return `(None, DELETED_METADATA)` and trying to get `None.metadata_type` will result in a crash.

Reply:
Oh, sorry, I see you already check for `r` existence ))